src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -525,6 +539,19 @@ internal IRResult ApplyIR()
                 this.ThrowOnErrors();
                 (var irnode, var ruleScopeSymbol) = IRTranslator.Translate(binding);
 
+                var list = _engine.IRTransformList;
+                if (list != null)
+                {
+                    foreach (var transform in list)
+                    {
+                        irnode = transform.Transform(irnode, _errors);
+
+                        // Additional errors from phases. 
+                        // Stop any further processing if we have errors. 
+                        this.ThrowOnErrors();
+                    }

```

Comment:
Wouldn't users generally want all the errors?

Reply:
IR is assuming we have valid input (we don't even try translation if binding fails, see 539). 

So to preserve that invariant, IR is always assuming we're valid, and hence any translation aborts if we get errors.  (warnings are ok). 
It makes it much easier to reason about now - although we could try to expand in the future if we care to get more errors.